pkg/m3admin/types.go

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package m3admin
+
+import (
+	"net/http"
+
+	"github.com/m3db/m3/src/query/generated/proto/admin"
+	"github.com/m3db/m3cluster/generated/proto/placementpb"
+)
+
+const (
+	M3coordinatorServiceName = "m3coordinator"
+	M3coordinatorPort        = 7201
+)
+
+type Namespace interface {
+	DefaultNamespaceRequest(string) *admin.NamespaceAddRequest
+	Create(string, string) (*http.Response, error)
```

Comment:
some nits:
- add comments to all method/interfaces
- add argument var names -- `(string, string)` doesn't convey much about how its going to be used
- why return http.Response here? how is it going to be used externally?

Reply:
To #1 and #2 Done. 

#3 the `http.Response` is returned to allow any consumer to act on the response of the call

Reply:
Re 3 - So it's leaky to expose the `http.Response` directly. Instead of this, could you expose a native go type for this instead and the do parsing to go from http.Response -> go type within the client. 

Will make usage of the interface simpler, and be a better abstraction. For e.g. if we change to use a GRPC endpoint instead of HTTP in the future, the code using these clients won't have to change.